syntax = "proto3";

package dictionary;

// Serviço gRPC
service DictionaryService {
  rpc ProcessWord (WordRequest) returns (WordResponse);
}

// Mensagens de Requisição e Resposta
message WordRequest {
  string word = 1;
}

message WordResponse {
  repeated WordEntry words = 1;
}

message WordEntry {
  string word = 1;
  int32 count = 2;
}
